functions/stdlib/at_quick_exit.c

```diff
@@ -0,0 +1,68 @@
+/* at_quick_exit( void (*)( void ) )
+
+   This file is part of the Public Domain C Library (PDCLib).
+   Permission is granted to use, modify, and / or redistribute at will.
+*/
+
+#include <stdlib.h>
+
+#ifndef REGTEST
+
+extern void (*_PDCLIB_quickexitstack[])( void );
+extern size_t _PDCLIB_quickexitptr;
+
+int at_quick_exit( void (*func)( void ) )
+{
+    if ( _PDCLIB_quickexitptr == _PDCLIB_ATEXIT_SLOTS )
+    {
+        return -1;
+    }
+    else
+    {
+        _PDCLIB_quickexitstack[ _PDCLIB_quickexitptr++ ] = func;
+        return 0;
+    }
+}
+
+#endif
+
+#ifdef TEST
+
+#include "_PDCLIB_test.h"
+
+#include <assert.h>
+
+static int flags[ 32 ];
+
+static void counthandler( void )
+{
+    static int count = 0;
+    flags[ count ] = count;
+    ++count;
+}
+
+static void checkhandler( void )
+{
+    int i;
+    for ( i = 0; i < 31; ++i )
+    {
+        assert( flags[ i ] == i );
+    }
+}
+
+int main( void )
+{
+#ifdef __ANDROID__
+    TESTCASE( NO_TESTDRIVER );
+#else
+    int i;
+    TESTCASE( at_quick_exit( &checkhandler ) == 0 );
+    for ( i = 0; i < 31; ++i )
+    {
+        TESTCASE( at_quick_exit( &counthandler ) == 0 );
+    }
+#endif
+    return TEST_RESULTS;
```

Comment:
Who calls the `quick_exit` function to run all handlers (and check)?
Is the C-compiler / linker aware of this by default - does *it* call the function by that name?

I'd have expected some "destructor" or platform specific code.

Reply:
I copied the testcase from `at_exit.c` and forgot to adapt it. I fixed and checked that it gets run when building and running tests.